### ISSUE TITLE
Movies with dots in name are badly detected when they start with the same component name

### ISSUE THREAD

**Issue by Dric** [Komet/MediaElch#1194]
I have multiples films with name components separated with dots. For example : 

```
Captain.Smart.The.Great.Bug.mkv
Captain.Pizza.The.Great.mkv
Captain.Pizza.II.The.Revenge.mkv
Captain.Y.Not.An.X.Movie.mkv

```
MediaElch is detecting only the first of these files, and the <baseFileName> is messed up : the nfo and jpeg files names are not identical to the movie file name.

Maybe it's by design, as the files are all grouped in one entry in MediaElch. Is there an option to disable this and get my movies detected separatedly ?

I'm on MediaElch 2.8.4 on Windows 10 x64, accessing the files from a SMB share.

**Comment by Dric** [comment]
Thanks for replying !

The files are on the same folder, next to each other.
All the generated files have the bad base name. It's generating one `Ca.nfo` for all the movies beginning with `Captain.`, and `Ca-fanart.jpeg`, `Ca-poster.jpeg`, etc.

I haven't enabled "Sep folder". (I dont really know what this option does...)

Regards,
Cédric

**Comment by bugwelle** [comment]
Thanks! I'll try to reproduce this issue.

> I haven't enabled "Sep folder". (I dont really know what this option does...)

Each movie has it's own directory. See https://kodi.wiki/view/Naming_video_files/Movies#Movie_Folder :-)

**Comment by bugwelle** [comment]
MediaElch sees all four movies as the same...

(Hover over the filename of the one movie that is recognized)

I'll look into why MediaElch does this...

**Comment by bugwelle** [comment]
Found the issue a man... that's an edge case.

For those that are interested in the technical issue:  
MediaElch tries to recognize movies that are split into multiple parts. We have this regular expression for it:

```
(.*)([ _.-]*(?:cd|dvd|p(?:ar)?t|dis[ck])[ _.-]*[a-d])(.*)(\.[^.]+)
`` `

And what do we get for `Captain.Smart.The.Great.Bug.mkv`? These capture groups:

2. "/Users/d067866/Projects/Private/MediaElch/scripts/generated_media/movies_flat/Ca"
3. "pta"
4. "in.Smart.The.Great.Bug"
5. ".mkv"

`pt` is matched. And that's why the rest is ignored. I'll see how I can fix it.

**Comment by bugwelle** [comment]
Will be fixed by #1197
